### PR TITLE
Sort out lint errors, tidy code

### PR DESCRIPTION
**Pull request by r-ash**
## Description

Apologies for this quite gross PR, this tidies up all lint errors or turns off the ones we were ignoring. 

Also clears unused imports, plus a few other bits of tidying

I started removing the `plottingMetadata` here too as it is unused, but then stopped on that because there is lots of handling in here about when that plottingMetadata is persisted. Figured that would be useful to do in 1 go when we add the plotMetadata into persisted state (if that is what we want to do)

## Type of version change
None

## Checklist

- [ ] I have incremented version number, or version needs no increment
- [ ] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/.eslintrc.js [hunk 1]
```diff
@@ -29,5 +29,16 @@ module.exports = {
...
+        "@typescript-eslint/no-explicit-any": "off",
+        // TODO: enable this again and fix occurances of non-null assert
+        "@typescript-eslint/no-non-null-assertion": "off",
+        "@typescript-eslint/no-unused-vars": [
+            "warn",
+            {
+                "argsIgnorePattern": "^_",
+                "varsIgnorePattern": "^_",
+                "caughtErrorsIgnorePattern": "^_"
+            }
```
Comment (r-ash) [inline src/app/static/.eslintrc.js:41]: With this if we have a deliberately unused var we can prefix it with `_` to make that clear

File: src/app/static/package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "copy": "cp -r ./templates/*.ftl ./public && cp -r ./assets/* ./public",
     "test": "vitest --run --coverage",
     "browser-test": "npx playwright test",
-    "integration-test": "vitest --config ./test-integration-configs/integration.ts --run --coverage",
+    "integration-test": "vitest --config test-integration-configs/integration.mts --run --coverage",
```
Comment (r-ash) [inline src/app/static/package.json:14]: File rename fixes a deprecation warning https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated

File: src/app/static/src/tests/apiService.test.ts [hunk 3]
```diff
@@ -276,7 +263,7 @@ describe("ApiService", () => {
             committedOptions = options;
         };
 
-        const response = await api({commit, rootState} as any)
+        const _response = await api({commit, rootState} as any)
```
Comment (M-Kusumgar) [inline src/app/static/src/tests/apiService.test.ts:266]: can also just remove the assignment and just do `await api...`

File: src/app/static/src/tests/components/app.test.ts [hunk 4]
```diff
@@ -117,7 +116,7 @@ describe("App", () => {
         document.body.appendChild(div);
 
         const store = getStore();
-        const rendered = shallowMount(Hint, {
+        const _rendered = shallowMount(Hint, {
```
Comment (M-Kusumgar) [inline src/app/static/src/tests/components/app.test.ts:119]: same here, can just remove the assignment

File: src/app/static/src/tests/components/modelOptions/modelOptions.test.ts [hunk 5]
```diff
@@ -222,7 +222,7 @@ describe("Model options component", () => {
             fetchModelRunOptions: fetchMock
 
         });
-        const rendered = shallowMount(ModelOptions, {
+        const _rendered = shallowMount(ModelOptions, {
```
Comment (M-Kusumgar) [inline src/app/static/src/tests/components/modelOptions/modelOptions.test.ts:225]: and here

File: src/app/static/src/tests/components/stepper.test.ts [hunk 6]
```diff
@@ -682,31 +684,31 @@ describe("Stepper component", () => {
     it("pushes router to projects if logged in user and currentProject not set", () => {
         const mockRouterPush = vi.fn();
         //current user is set in vi.config and currentProject is not set be default in the wrapper
-        const wrapper = createSut({}, {}, {}, {}, {}, {}, {}, mockRouterPush);
+        const _wrapper = createSut({}, {}, {}, {}, {}, {}, {}, mockRouterPush);
```
Comment (M-Kusumgar) [inline src/app/static/src/tests/components/stepper.test.ts:687]: can get rid of all the assignments in this file

File: src/app/static/src/tests/stepper/stepperState.test.ts [hunk 7]
```diff
@@ -10,7 +8,8 @@ const existingState = {
         selectedDataType: null
     },
     baseline: {}
-} as any;;
+} as any;
+;
```
Comment (M-Kusumgar) [inline src/app/static/src/tests/stepper/stepperState.test.ts:12]: i think we can just delete this second one :laughing:

**Pull request merged**